### ISSUE TITLE
Move local directory / folder without re-downloading

### ISSUE THREAD

**Issue by Zocker1999NET** [nextcloud/desktop#2460]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Note
Thought a similar issue might exist, but did not found any one after searching for 10 minutes, so here is this feature request. Sorry if there was already an similar one.

### Expected behaviour
Want to move my local synced directory from `~/Nextcloud` to `~/MyCloud` **without** re-downloading all files. There should be a button in the client to move.
Even if it would work, I expect that the client will re-download all files or, worse, delete all files on remote if I move the directory without using the client.

### Actual behaviour
Did not found any button.

### Workaround
For anyone who cannot wait and has no problem with re-downloading all files again: You can unsync the local directory and resync the files to a new location by adding a new synced directory, but I hope it might be possible without re-syncing.

### Steps to reproduce
1. Open Nextcloud Desktop by left-clicking on the icon in the info tray.
2. Click on the three dots right of the local folder / directory I want to move.
3. Found no button "move folder" (etc)

### Client configuration
Client version: 2.6.4-1

Operating system: Debian Testing/Unstable

OS language: English

Qt version used by client package (Linux only, see also Settings dialog): did not found any Qt version in the settings dialog, might be newest available for Debian Testing right now: 
![image](https://user-images.githubusercontent.com/1645646/93867071-d4ad3000-fcc8-11ea-801f-e00f9dcc97f6.png)

Client package (From Nextcloud or distro) (Linux only): distro

Installation path of client: see distro

**I expect the server version & logs not to be required, if required, I can still provide these information.**

**Comment by er-vin** [comment]
Just to be sure, this is about moving files and folders between two sync folders on the same server?

**Comment by Zocker1999NET** [comment]
This is about moving the same synced folder from the same server **locally** from one location to another.

**Comment by er-vin** [comment]
Sorry I badly worded this. Let me try again did I get it right that you got the following setup:
 * client configured with a single account;
 * two sync folders for that single account, one pointing to `~/Nextcloud` the other one to `~/MyCloud`;
 * you're trying to move files from `~/Nextcloud` to `~/MyCloud` and want the move to be replicated server side and not with a remove + add.

Is this correct?

**Comment by Zocker1999NET** [comment]
@er-vin, no. Sorry, my description must be bad too. So I try to make an example too:

Client is configured with a single account and only configured `~/Nextcloud` to be a sync folder. Now I want to move these offline copies to `/mnt/hdd/Nextcloud` because my home/root drive (`~`) lacks storage but my additional HDD (`/mnt/hdd`) has plenty of storage. The server is not here involved at all and should not notice the change. Because I have a bad internet connection, I do not want to "unsync" the folder `~/Nextcloud` and re-download all files to `/mnt/hdd/Nextcloud`, but I want `/mnt/hdd/Nextcloud` to be a new replica of my cloud files while `~/Nextcloud` being removed/removeable.

**Comment by er-vin** [comment]
So, let me rephrase to see if I got it right this time: you want to move the local sync folder from one path to another?

So there are ways to do this behind the back of the client but you'd like this to be a feature from the client? (hence why you mentioned a button earlier)

**Comment by Zocker1999NET** [comment]
@er-vin Correct! I suggested to add a button because, like you mentioned, it might be possible to this from the background, but
1. IMO most users will think that just moving the directory with OS tools without saying this to the client might result into a re-download
2. Doing this the right way either requires user to follow a complex set of instructions (complex for normal users) or, worse, insight in how the client does work.
A button which will do this for the user might be the perfect solution.

If possible, the button could also say that moving the directory with OS tools (e.g. the default file explorer) will work as long as the client is running and, of course, support this (OneDrive on Windows 10 claims supporting this if I remember correctly), but I think this might be hard to implement in a safe way.

**Comment by er-vin** [comment]
Thanks for the clarification, not sure we'll do this soon though. We're taking patches obviously, this could be interesting.

**Comment by kitatek** [comment]
> Thanks for the clarification, not sure we'll do this soon though. We're taking patches obviously, this could be interesting.

I added the 👍🏼 as requested but I think this comment will help prioritizing: I think adding this feature can help Nextcloud take more ground in the corporate IT field.

The reason is many corporate IT support teams or providers have the habit of splitting the client disks into two partitions: C:\ system for Windows, and D:\ data for documents.

Unfortunately, most users install with defaults and fill their smaller  C:\ drive with synced docs.

Adding this move feature will make their lives much easier with their Nextcloud experience.

**Comment by burner-** [comment]
I totally agree that this is needed option. I have no idea what happens if I go to os and try move that folder. I assume that it wil blow up many ways. At least drop box end up to lost and corrupt some of files and lot of other problems.

**Comment by eighthave** [comment]
I moved a sync account from _~/Nextcloud2_ to _~/Documents_.  My workaround was this:
```console
$ killall nextcloud
$ sed -i 's,Nextcloud2,Documents,g' \
    ~/.config/Nextcloud/nextcloud.cfg ~/.config/gtk-3.0/bookmarks
```

Seems to work so far.

**Comment by joshtrichards** [comment]
Duplicate of #2161
